crates/anvil-polkadot/tests/it/snapshot.rs

```diff
@@ -0,0 +1,598 @@
+    assert_eq!(node.get_nonce(alith_addr).await, U256::ZERO);
+    assert_eq!(node.get_nonce(baltathar_addr).await, U256::ZERO);
+    assert_eq!(node.get_nonce(dest_addr).await, U256::ZERO);
+
+    // Remine the 6th block with same txs above.
+    let transaction =
+        TransactionRequest::default().value(U256::from(16e17)).from(alith_addr).to(dest_addr);
+    let tx_hash1 = node.send_transaction(transaction, None).await.unwrap();
+    let transaction =
+        TransactionRequest::default().value(U256::from(1e17)).from(baltathar_addr).to(alith_addr);
+    let tx_hash2 = node.send_transaction(transaction, None).await.unwrap();
+    unwrap_response::<()>(node.eth_rpc(EthRequest::Mine(Some(U256::from(1)), None)).await.unwrap())
+        .unwrap();
+    assert_block_number(&mut node, 6).await;
+
+    tokio::time::sleep(std::time::Duration::from_millis(500)).await;
+    let receipt_info = node.get_transaction_receipt(tx_hash1).await;
+    assert_eq!(receipt_info.block_number, pallet_revive::U256::from(6));
+    assert_eq!(receipt_info.transaction_index, pallet_revive::U256::one());
+    assert_eq!(receipt_info.transaction_hash, tx_hash1);
+    let receipt_info = node.get_transaction_receipt(tx_hash2).await;
+    assert_eq!(receipt_info.block_number, pallet_revive::U256::from(6));
+    assert_eq!(receipt_info.transaction_index, pallet_revive::U256::from(2));
+    assert_eq!(receipt_info.transaction_hash, tx_hash2);
+    assert_eq!(node.get_nonce(alith_addr).await, U256::ONE);
+    assert_eq!(node.get_nonce(baltathar_addr).await, U256::ONE);
+    assert_eq!(node.get_nonce(dest_addr).await, U256::ZERO);
+}
+
+#[tokio::test(flavor = "multi_thread")]
+// TODO: add a test where we call a contract that queries the timestamp
+// at a certain block before and after a revert, while mining blocks
+async fn test_evm_revert_and_timestamp() {
+    let anvil_node_config = AnvilNodeConfig::test_config().with_no_mining(true);
+    // Generate the current timestamp and pass it to anvil config.
+    let genesis_timestamp = anvil_node_config.get_genesis_timestamp();
+    let anvil_node_config = anvil_node_config.with_genesis_timestamp(Some(genesis_timestamp));
+    let substrate_node_config = SubstrateNodeConfig::new(&anvil_node_config);
+    let mut node = TestNode::new(anvil_node_config.clone(), substrate_node_config).await.unwrap();
+
+    // Do a first snapshot for genesis.
+    let id = unwrap_response::<String>(node.eth_rpc(EthRequest::EvmSnapshot(())).await.unwrap())
+        .unwrap();
+    assert_eq!(id, "0x0".to_string());
+
+    // Assert on first best block number.
+    unwrap_response::<()>(node.eth_rpc(EthRequest::Mine(Some(U256::from(1)), None)).await.unwrap())
+        .unwrap();
+    assert_eq!(node.best_block_number().await, 1);
+    let first_timestamp = node.get_decoded_timestamp(None).await;
+    assert_with_tolerance(
+        first_timestamp.saturating_div(1000).saturating_sub(genesis_timestamp),
+        0,
```

Comment:
it's more intuitive to compare these two rather than comparing their difference against 0.

Also, a 1 second tolerance is a bit large?